app/src/org/commcare/views/EntityActionViewUtils.java

```diff
@@ -64,6 +66,9 @@ private static void setupActionClickListener(FrameLayout actionCardView,
                                                  final Action action,
                                                  final CommCareActivity commCareActivity) {
         CardView cardView = actionCardView.findViewById(R.id.card_body);
-        cardView.setOnClickListener(v -> EntitySelectActivity.triggerDetailAction(action, commCareActivity));
+        cardView.setOnClickListener(v -> {
+            cardView.setEnabled(false);
```

Comment:
Extremely minor point: I don't think this is a real concern, but if the EntityList view was recycled entirely (IE: The use pressed 'back' and the view came up exactly as it was) it would be the case that this button would remain disabled. 

Again, I don't think this applies to the way we use the views, but I didn't want to discount it entirely in case you knew differently, since you have worked with this more recently than  I have.

Reply:
@ctsims Thanks for flagging. You are right that it's a non concern as `onClick` here calls `EntitySelectActivity.triggerDetailAction` which is a terminal operation in the sense that it always `finish` the current activity. So pressing back always results in CommCare session to recreate the `EntityListActivity` again re-enabling the action button.

Reply:
great, thanks for confirming